.clang-tidy

```diff
@@ -0,0 +1,13 @@
+---
+Checks:                 'clang-diagnostic-*,clang-analyzer-*,boost-*,bugprone-*,cert-*,cppcoreguidelines-*,-cppcoreguidelines-macro-usage,-cppcoreguidelines-avoid-magic-numbers,-cppcoreguidelines-c-copy-assignment-signature,-cppcoreguidelines-pro-type-reinterpret-cast,hicpp-*,-hicpp-explicit-conversions,misc-*,-misc-unused-using-decls,performance-*,portability-*'
+WarningsAsErrors:       ''
+HeaderFilterRegex:      ''
+AnalyzeTemporaryDtors:  false
+FormatStyle:            none
```

Comment:
Isn't there a particular format style that matches what we're doing? It'd be great if we could have incoming PRs tested against our coding style.

Reply:
See discussion below under `tests/unit_tests/signature.cpp`.